iib/workers/tasks/build.py

```diff
@@ -720,6 +721,66 @@ def _verify_index_image(
         )
 
 
+def _add_property_to_index(db_path, property):
+    """
+    Add a property to the index.
+
+    :param str db_path: path to the index database
+    :param dict property: a dict representing a property to be added to the index.db
+    """
+    insert = 'INSERT INTO properties (type, value, operatorbundle_name, operatorbundle_version, \
+        operatorbundle_path) VALUES (?, ?, ?, ?, ?);'
+    con = sqlite3.connect(db_path)
+    # Insert property
+    con.execute(
+        insert,
+        (
+            property['type'],
+            property['value'],
+            property['operatorbundle_name'],
+            property['operatorbundle_version'],
+            property['operatorbundle_path'],
+        ),
+    )
+    con.commit()
+    con.close()
+
+
+def _requires_max_ocp_version(bundle):
+    """
+    Check if the bundle requires the olm.maxOpenShiftVersion property.
+
+    This property is required for bundles using deprecated APIs that don't already
+    have the olm.maxOpenShiftVersion property set.
+
+    :param str bundle: a string representing the bundle pull specification
+    :returns bool:
+    """
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1',
+        '-b',
+        'podman',
+    ]
+    result = run_cmd(cmd, strict=False)
+    if result:
+        output = json.loads(result)
+        # check if the bundle validation failed
+        if not output['passed']:
+            # check if the failure is due to the presence of deprecated APIs
+            # and absence of the 'olm.maxOpenShiftVersion' property
+            # Note: there is no other error in the sdk that mentions this field
+            for msg in output['outputs']:
+                if 'olm.maxOpenShiftVersion' in msg['message']:
+                    return True
+    return False
```

Comment:
If the `output['passed']` returns `true`, so this `return False` will be run, I guess it's not as expected, or am I missing something? Thanks!

Reply:
If the bundle passes the set of tests then it does not require (or the operator-sdk cannot detect that it requires) the `olm.maxOpenShiftVersion` label

Reply:
Yes, thanks! Sorry, my misunderstanding.